docgen/src/guide/Refreshing_cache.md

```diff
@@ -0,0 +1,115 @@
+
+
+There are two different use cases where you would want to discard the cache:
+- your application data is being directly updated by your users (for example, in a dashboard). In this use case you would want to refresh the cache based on some application state such as the last modification from the user
+- your application data is being updated by another process that you don't manage (for example a CRON job updates users inside Algolia). For this you might want to periodically refresh the cache of your application
+
+
+ 
+ ## Have the refresh of the cache triggered by an action from the user
+If you know that the cache needs to be refreshed conditionally of a specific event, then you can trigger the refresh based on an user action (adding a new product, clicking on a button for instance).
+
+```jsx
+import React, { Component } from 'react';
+import { InstantSearch, SearchBox } from 'react-instantsearch/dom';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      refresh: false,
+    };
+  }
+
+  refresh = () => {
+    this.setState({refresh: true});
+  };
+
+  onSearchStateChange = () => {
+    this.setState({ refresh: false });
+  };
+
+  render() {
+    return (
+      <InstantSearch
+        appId="yourAppId"
+        apiKey="yourApiKey"
+        indexName="yourIndexName"
+        refresh={this.state.refresh}
+        onSearchStateChange={this.onSearchStateChange}
+      >
+        <SearchBox />
+        <button
+          onClick={this.refresh}
+        >
+          Refresh cache
+        </button>
+        <CustomHits />
+      </InstantSearch>
+    );
+  }
+}
+```
+
+See a live example on our [Storybook](https://community.algolia.com/react-instantsearch/storybook/?selectedKind=RefreshCache&selectedStory=with%20a%20refresh%20button&full=0&down=1&left=1&panelRight=1&downPanel=storybooks%2Fstorybook-addon-knobs). 
+
+ ## Refresh the cache periodically
+You also have the option to setup an given period of time that will determine how often the cache will be cleared. 
+This method will ensure that the cache is cleared on a regular basis.
+You should use this approach if you cannot use a user action as a specific event to trigger the clearing of the cache.
+
+```jsx
+import React, { Component } from 'react';
+import { InstantSearch, SearchBox } from 'react-instantsearch/dom';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      refresh: false,
+    };
+  }
+
+  componentDidMount = () => {
+    setInterval(
+      () =>
+        this.setState({ refresh: true }), 5000 );
+  };
+
+   onSearchStateChange = () => {
+    this.setState({ refresh: false });
+  };
+  render() {
+    return (
+      <InstantSearch
+        appId="yourAppId"
+        apiKey="yourApiKey"
+        indexName="yourIndexName"
+        refresh={this.state.refresh}
+        onSearchStateChange={this.onSearchStateChange}
+      >
+        <SearchBox />
+        <CustomHits />
```

Comment:
This variable doesn't exist in the snippet, use `Hits` instead of `CustomHits`. Don't forget to add the import.